.github/workflows/api-build-deploy.yaml

```diff
@@ -0,0 +1,113 @@
+        run: |
+          docker login ${{ secrets.ARTIFACTORY_URL }} -u ${{ secrets.ARTIFACTORY_USERNAME }} -p ${{ secrets.ARTIFACTORY_PASSWORD }}
+
+      - name: Copy env.template
+        run: |
+          cp .env.template .env
+      
+      - name: Build API Image
+        run: |
+          docker-compose build spr-api
+
+      - name: List Docker Images
+        run: |
+          docker images
+        
+      - name: Tag Image
+        run: |
+          docker tag citz-imb-staff-purchasing-reimbursement_spr-api ${{ secrets.ARTIFACTORY_URL }}/${{ secrets.IMAGE_REPOSITORY }}/spr-api:latest
+      
+      - name: Push Image
+        run: |
+          docker push ${{ secrets.ARTIFACTORY_URL }}/${{ secrets.IMAGE_REPOSITORY }}/spr-api:latest
+  
+  deploy:
+    name: Deploy to OpenShift
+    runs-on: ubuntu-latest
+    needs: [ build-tag-push ]
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Login to Openshift
+        uses: redhat-actions/oc-login@v1
+        env:
+          OPENSHIFT_USER: github-actions
+          OPENSHIFT_PROJECT: ${{ secrets.OPENSHIFT_PROJECT }}
+        with: 
+          openshift_server_url: ${{ secrets.OPENSHIFT_SERVER_URL }}
+          openshift_token: ${{ secrets.OPENSHIFT_TOKEN }}
+          namespace: ${{ secrets.OPENSHIFT_PROJECT }}
+
+      - name: Deploy from DeploymentConfig
+        env:
+          OPENSHIFT_PROJECT: ${{ secrets.OPENSHIFT_PROJECT }}
+          APPLICATION_NAME: staff-purchase-reimbursement
+          CONTAINER_NAME: spr-api
+          DEPLOYMENT_CONFIG: api-deployment-config.yaml
+          IMAGE_REPOSITORY: ${{ secrets.IMAGE_REPOSITORY }}
+          ARTIFACTORY_URL: ${{ secrets.ARTIFACTORY_URL }}
+        run: |
+          ./.github/helpers/deploy.sh
+
+  healthcheck:
+    name: Determine Rollout Success
+    runs-on: ubuntu-latest
+    needs: [ deploy ]
+    steps:
+      - name: Login to Openshift
+        uses: redhat-actions/oc-login@v1
+        env:
+          OPENSHIFT_USER: github-actions
+          OPENSHIFT_PROJECT: ${{ secrets.OPENSHIFT_PROJECT }}
+        with: 
+          openshift_server_url: ${{ secrets.OPENSHIFT_SERVER_URL }}
+          openshift_token: ${{ secrets.OPENSHIFT_TOKEN }}
+          namespace: ${{ secrets.OPENSHIFT_PROJECT }}
+      
+      - name: Check Status of Rollout
+        run: |
+          oc rollout status -n ${{ secrets.OPENSHIFT_PROJECT }} dc/spr-api --watch
+  
+  cleanup:
+    name: Clean Up Old Objects
+    runs-on: ubuntu-latest
+    needs: [ healthcheck ]
+    steps:
+      - name: Login to Openshift
+        uses: redhat-actions/oc-login@v1
+        env:
+          OPENSHIFT_USER: github-actions
+          OPENSHIFT_PROJECT: ${{ secrets.OPENSHIFT_PROJECT }}
+        with: 
+          openshift_server_url: ${{ secrets.OPENSHIFT_SERVER_URL }}
+          openshift_token: ${{ secrets.OPENSHIFT_TOKEN }}
+          namespace: ${{ secrets.OPENSHIFT_PROJECT }}
+
+      - name: Remove old pod objects
```

Comment:
could refactor these into a single cleanup helper script later if you want

Reply:
I saw that was what DPIA had done. I was felt that this was more transparent as to what was happening since they were only one-liners.